pepper-apis/src/main/java/org/broadinstitute/ddp/route/AddProfileRoute.java

```diff
@@ -32,19 +33,14 @@ public Object handle(Request request, Response response, Profile profile) throws
         String userGuid = request.params(RouteConstants.PathParam.USER_GUID);
         LOG.info("Creating profile for user with guid {}", userGuid);
 
-        if (profile == null) {
-            ResponseUtil.halt400ErrorResponse(response, ErrorCodes.MISSING_BODY);
-            return null;
-        }
-
         UserProfile.SexType sex = null;
         if (profile.getSex() != null) {
             try {
                 sex = UserProfile.SexType.valueOf(profile.getSex());
             } catch (IllegalArgumentException e) {
                 LOG.warn("Provided invalid profile sex type: {}", profile.getSex(), e);
-                ResponseUtil.halt400ErrorResponse(response, ErrorCodes.INVALID_SEX);
-                return null;
+                throw ResponseUtil.haltError(400, new ApiError(ErrorCodes.INVALID_SEX,
```

Comment:
Let's use the provided constants for the HTTP status codes instead.

```suggestion
                throw ResponseUtil.haltError(HttpStatus.SC_BAD_REQUEST, new ApiError(ErrorCodes.INVALID_SEX,
```